python/kvikio/kvikio/_lib/future.pxd

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
+
+# distutils: language = c++
+# cython: language_level=3
+
+
+cdef extern from "<future>" namespace "std" nogil:
+    cdef cppclass future[T]:
```

Comment:
If there is an issue requesting std::future in Cython, please link it in a comment here. If there is no issue, please file one and then link it.